docs/design/DIDManagement.md

```diff
@@ -0,0 +1,186 @@
+- **Value:** The value stored in the entry. This is usually a serialized JSON object.
+- **Tags:** A mapping of strings to strings or lists of strings. These values can be used with the ["Wallet Query Language (WQL)"](https://github.com/hyperledger-archives/indy-sdk/tree/main/docs/design/011-wallet-query-language) to look up encrypted Askar entries efficiently.
+
+Askar has a dedicated API for storage and retrieval of keys. However, this API is conceptually just a shorthand for record storage and retrieval from a "private" `key` category with the key itself as the value of the entry. Key entries behave almost exactly the same as non-key entries, including names and tags.
+
+### Current State of DID Storage
+
+At present, the `DIDInfo` class in ACA-Py is structured as follows:
+
+```python
+DIDInfo = NamedTuple( 
+    "DIDInfo", 
+    [ 
+        ("did", str), 
+        ("verkey", str), 
+        ("metadata", dict), 
+        ("method", DIDMethod), 
+        ("key_type", KeyType), 
+    ], 
+) 
+```
+
+When stored in Askar, DID records have the following characteristics:
+
+- **Category:** did
+- **Name:** the value of the did, e.g. `did:example:123`. For Indy/did:sov, the value is the nym, e.g. `As728S9715ppSToDurKnvT`
+- **Value:** a JSON object with the following attributes:
+    - `did`: the DID (or nym)
+    - `method`: the method name, e.g. `peer`
+    - `verkey`: the base58 encoding of the public key associated with this DID
+    - `verkey_type`: the key type of the verkey, e.g. `ed25519`
+    - `metadata`: A container for arbitrary metadata. In practice, the following values are inserted into metadata:
+        - `posted`: a boolean value representing whether this DID has been published to an indy network
+        - `endpoint`: a string value representing the endpoint attrib of this DID on an indy network
+- **Tags:**
+    - `method`: the method name, e.g. `peer`
+    - `verkey`: the base58 encoding of the public key associated with this DID
+    - `verkey_type`: the key type of the verkey, e.g. `ed25519`
+
+### Current State of Key Storage
+
+Keys are managed by using the "verkey" as the name of the Askar record. Operations like signing or encrypting DIDComm messages retrieve the key by verkey. Usually, when initiating a cryptographic operation, the key is looked up by first retrieving the `DIDInfo` object by DID (or by nym) from the wallet and then the `DIDInfo.verkey` value is used to retrieve the key.
+
+### Limitations
+
+- **Indy-Centric Design:** The current structure is closely tied to Indy, making it less suitable for other DID Methods.
+- **Single Key per DID:** Assumes a one-to-one relationship between DIDs and keys, which is not the case for many DID Methods.
+- **Inefficient for Multiple Keys:** Lacks support for DIDs with multiple verification methods and keys.
+
+## Proposed Updates
+
+### Goals
+
+- **Support Multiple Keys per DID:** Allow DIDs to have multiple associated keys and verification methods.
```

Comment:
Should we add that we would like to track whether a key is active or historical/archived/no longer in use.  When we add `did:webvh` we will have keys referenced only in older versions of the DID/DIDDoc that should no longer be used for signing.

Reply:
Interesting question... If they should no longer be used for signing, do we have a need to keep the private key material around? If yes, then marking a key as inactive makes sense. If not, we could just remove the key from storage.